### ISSUE TITLE
False positive (already solved in dev-master)

### ISSUE THREAD

**Issue by ErikBooijCB** [vimeo/psalm#775]
The following code results in an error in the current 2.0 release (and prior ones). It's already fixed in dev-master, so it would be very sweet to see a minor update somewhere in the near future :)

```
TypeDoesNotContainType - src/test.php:37:13 - Values int(1) and int(2) can never be identical
```

https://getpsalm.org/r/6abf7f36a6

```php
<?php
declare(strict_types=1);

interface EventProviderInterface {}

class SourceAEventProvider implements EventProviderInterface {}
class SourceBEventProvider implements EventProviderInterface {}
class CombinedEventProvider implements EventProviderInterface {}

abstract class AbstractEventProvider {
    /** @var int */
    const SomeConstant = 1;

    /** @var int */
    const SomeOtherConstant = 2;
}

class EventProviderFactory
{
    /**
     * @param int $eventStore
     *
     * @return EventProviderInterface
     */
    public function createEventProviders(int $eventStore)
    {
        $providers = [];

        if (($eventStore & AbstractEventProvider::SomeConstant) === AbstractEventProvider::SomeConstant) {
            $providers[] = new SourceAEventProvider();
        }

        if (($eventStore & AbstractEventProvider::SomeOtherConstant) === AbstractEventProvider::SomeOtherConstant) {
            $providers[] = new SourceBEventProvider();
        }

        if (count($providers) === 1) {
            return array_shift($providers);
        }

        return new CombinedEventProvider();
    }
}
```

**Comment by muglug** [comment]
https://github.com/vimeo/psalm/releases/tag/2.0.1
